creusot-contracts/src/logic/eq.rs

```diff
@@ -0,0 +1,22 @@
+use crate as creusot_contracts;
+use creusot_contracts_proc::*;
+
+pub trait EqLogic {
```

Comment:
Do we require `log_eq` to be extensional?

Reply:
If so, do we need any other laws?

Reply:
Do you mean by "extensional" that it should be equivalent to `===`? I don't think this is a great idea. For some types, we really need an equality which is not `===`.

If you want, we can also have a trait for types where `===` coincides with `==`.

Reply:
Yea that's what I meant but it's probably better to separate the two like `PartialOrd` and `Ord` are